backend/test_observer/data_access/models.py

```diff
@@ -109,9 +109,8 @@ class Artefact(Base):
     status: Mapped[ArtefactStatus | None]
 
     __table_args__ = (
-        UniqueConstraint(
-            "name", "version", "track", "series", "repo", name="unique_artefact"
-        ),
+        UniqueConstraint("name", "version", "track", name="unique_snap"),
```

Comment:
could you please tell me what this "name" field mean? how does the DB know that it's related only to snaps?

Reply:
"name" is the name of the Artefact. Unless you mean `name="unique_snap"` which means the name of the constraint on the database.

The way this works all depends on how PostgreSQL handles null values in a unique constraint. Basically, if any of the values in the constraint is null then postgresql would essentially say this whole record is unique, even if other the values already exists. For example, if two artefacts had the same `name` and `version` but had null track then postgresql doesn't complain. But if they also had the same track then postgresql would not allow it. That means the the `unique_snap` constraint doesn't do anything when the artefact is a deb because track would be null. Instead, only the `unique_deb` constraint stops duplicates. Similarly, the `unique_deb` constraint doesn't do anything when the artefact is a snap because both series and repo are null. Instead, only the `unique_snap` constraints stops duplicate.